### ISSUE TITLE
Log attempts to regain modelock and try a bit harder.

### ISSUE THREAD

**Issue by raacampbell** [SainsburyWellcomeCentre/BakingTray#184]
BakingTray could be a lot more robust to failure of the laser. It should pause the acquisition and try harder to regain modelocking.

**Comment by raacampbell** [comment]
Further, sections where the laser failed do not have the COMPLETED file.

**Comment by raacampbell** [comment]
FYI @ablot 
Modelock failure is tested after the tile scan. If the laser is not modelocked once the tile scan is complete, COMPLETED file is not created and the `bake` method ends prematurely (although it will run its "cleanup" function). Therefore when the acquisition is resumed the section with missing tiles will be re-imaged. The sections associated with modelock failures are logged in the main log file in the sample root directory.

**Comment by raacampbell** [comment]
The messages are, oddly, that the laser has turned off. Not that it's lost modelock. So to be sure I add code to explicitly check this. Then add code to try turning it back on again.

**Comment by raacampbell** [comment]
I finish acquisition by selecting "stop after section" and I get odd errors that make no sense:
<img width="1440" alt="Screenshot 2019-05-02 at 01 08 57" src="https://user-images.githubusercontent.com/5427811/57050979-0c9c7900-6c77-11e9-849a-564ec65ce841.png">
Unless it's just chance that the laser turns off at this point, this is most odd.

**Comment by raacampbell** [comment]
To do:
* [ ] Monitor tower temp and report if above 25 degrees. 
* [ ] Laser returns status codes indicating previous errors. Set up a system for reporting those. Either when there is a failure or at some other time.

**Comment by ablot** [comment]
Thanks.
I assumed that the reimaged section would be the next section (i.e. if the laser failed in the middle of section 70, I would have a folder `section_70` with half of the tiles being black and and a `section_71` folder with the same section imaged when I restarted the `bake`). 
Is that right?

**Comment by raacampbell** [comment]
I think that is correct. But the stitching doesn't know about that.

Rob

**Comment by raacampbell** [comment]
At least based on the Slack messages, it seems that laser modelock failures are now being caught and fixed on many occasions (50ea51ceee0a0c0d8ad9f3256ec8153dbb91f1a4), although I don't know for sure that the laser had lost modelock. Maybe it's lying. 

The repeated failures we saw previously were due to the laser getting too warm. In one instance the light shroud curtain was blocking the intake vents one the chiller. This is no longer possible.
